### PR TITLE
Fix transmute mess

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/gtk-rs/gir/pull/902

cc @EPashkin @sdroege

### CODE REVIEW

File: src/object.rs [hunk 1]
```diff
@@ -1541,7 +1541,7 @@ impl<T: ObjectType> ObjectExt for T {
         ::signal::connect_raw(
             self.as_object_ref().to_glib_none().0,
             signal_name.as_ptr() as *const _,
-            Some(*(&notify_trampoline::<Self, F> as *const _ as *const _)),
+            Some(mem::transmute(notify_trampoline::<Self, F> as usize)),
```
Comment (sdroege) [inline src/object.rs:1544]:
```suggestion
            Some(mem::transmute::<_, unsafe extern "C" fn()>(notify_trampoline::<Self, F> as *const ())),
```

etc. Let's not introduce unbounded transmutes again.

File: src/source.rs [hunk 2]
```diff
@@ -611,7 +612,7 @@ where
         let source = glib_sys::g_child_watch_source_new(pid.0);
         glib_sys::g_source_set_callback(
             source,
-            Some(*(&trampoline_child_watch::<F> as *const _ as *const _)),
+            Some(transmute(trampoline_child_watch::<F> as usize)),
```
Comment (sdroege) [inline src/source.rs:615]:
```suggestion
            Some(transmute::<_, unsafe extern "C" fn(glib_sys::gpointer) -> glib_sys::gboolean>(trampoline_child_watch::<F> as *const ())),
```

File: src/source.rs [hunk 3]
```diff
@@ -678,7 +679,7 @@ where
         let source = glib_sys::g_unix_fd_source_new(fd, condition.to_glib());
         glib_sys::g_source_set_callback(
             source,
-            Some(*(&trampoline_unix_fd::<F> as *const _ as *const _)),
+            Some(transmute(trampoline_unix_fd::<F> as usize)),
```
Comment (sdroege) [inline src/source.rs:682]:
```suggestion
            Some(transmute::<_, unsafe extern "C" fn(glib_sys::gpointer) -> glib_sys::gboolean>(trampoline_unix_fd::<F> as *const ())),
```

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by sdroege** [comment]
Minimum Rust version has to be updated to 1.40 for the `#[non_exhaustive]` attribute

**Pull request merged**